### ISSUE TITLE
Generated files use LF on windows, should use `require('os').EOL`

### ISSUE THREAD

**Issue by JeroMiya** [cyclosproject/ng-openapi-gen#77]
Not sure if this is an issue in ng-openapi-gen code or one of its dependencies, but currently the files it generates use `LF` line endings on Windows. When using `core.autocrlf true` on Windows, the generated files in the repo become `CRLF` after checkout, then when the services and models are regenerated every file shows up as changed (since it looks like `CRLF` -> `LF`).

Solution would be something along the lines of using `require('os').EOL` for the line endings instead of assuming `LF`. Again, not sure if the issue is in ng-openapi-gen or a dependency.

**Comment by luisfpg** [comment]
Actually, it is a bad practice to add generated files to git.
There are several discussions on the topic, such https://kentcdodds.com/blog/why-i-dont-commit-generated-files-to-master or https://homes.cs.washington.edu/~mernst/advice/version-control.html#Dont_commit_generated_files
If you really want to commit generated files, you could have a npm script that generates and then replaces the LF chars by CRLF, such as https://github.com/sindresorhus/normalize-newline-cli.

**Comment by JeroMiya** [comment]
If it's ok, I'd really like to request that this issue be re-opened. I do not consider it invalid. All files generated by an npm based tool should match the line endings of the platform they run on whether they are checked into git or not.

Also, for what it's worth, I disagree that it is bad practice to check in generated code or other files to git for two practical reasons:
1. Code generators often have bugs. In many cases bugs for which there is no working version of the tool to fall back to. If generated files are checked into git, buggy output can be hand-fixed temporarily while the generator bug is being fixed. The alternatives are to switch to another tool (if one exists), which is disruptive, wait until the bug is fixed, which is unsustainable, or temporarily fork the tool and fix it ourselves (which takes time away from the original project). 
2. Checking generated code into git allows you to audit line by line differences caused by an updated version of the generator or an updated version of the input files to the generator. If software quality is a goal, these change audits are critical to that goal. Two of the bugs I've filed for ng-openapi-gen were the result of such an audit.

I have already implemented my own tooling to automatically fix the line endings in the generated output files, but I consider this a workaround more than a solution.

**Comment by luisfpg** [comment]
Let's suppose that I, which use Linux, join your development team. So, every time I generate the code all line endings would be wrong anyway.
So, using the platform line endings is not really a solution.
Anyway, I'm reopening it, because could work with small / homogeneous teams.

**Comment by JeroMiya** [comment]
Linux and Mac users would be fine. Git will checkout files with the LF line endings on those platforms, and if the tool uses the platform line endings, it will generate LF line endings and you'll be golden. We've got a mix of mac/linux/windows users on this team so it's a constant struggle when npm tools write the wrong line endings :D Thanks for re-opening!

**Comment by luisfpg** [comment]
@JeroMiya  Can you, please, test the commit https://github.com/cyclosproject/ng-openapi-gen/commit/8422f5860e13f36307f884ad9d0c0fdefc877431?
You'll have to clone, run:
```bash
npm run build
cd dist
npm link
# Then, in your project
npm link ng-openapi-gen
npm start
```

**Comment by luisfpg** [comment]
@JeroMiya Actually I've published the 0.8.2 release.
If you still have problems, please, file a new issue.

**Comment by tmeinhart** [comment]
I'm working on a windows machine and i'm facing the described problem, with the difference, that in the generated code the summary of a function is created with LF as line ending, while the rest of the code is generated using CRLF as line ending. Am i missing some configurations here?
